examples/post_training_quantization/torch/fx/resnet18/main.py

```diff
@@ -0,0 +1,231 @@
+
+    val_dataset = datasets.ImageFolder(
+        val_dir,
+        transforms.Compose(
+            [
+                transforms.Resize(IMAGE_SIZE),
+                transforms.ToTensor(),
+                normalize,
+            ]
+        ),
+    )
+
+    val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=1, shuffle=False, num_workers=0, pin_memory=True)
+
+    calibration_dataset = torch.utils.data.DataLoader(
+        val_dataset, batch_size=1, shuffle=False, num_workers=0, pin_memory=True
+    )
+
+    return val_loader, calibration_dataset
+
+
+def prepare_tiny_imagenet_200(dataset_dir: Path):
+    # Format validation set the same way as train set is formatted.
+    val_data_dir = dataset_dir / "val"
+    val_images_dir = val_data_dir / "images"
+    if not val_images_dir.exists():
+        return
+
+    val_annotations_file = val_data_dir / "val_annotations.txt"
+    with open(val_annotations_file, "r") as f:
+        val_annotation_data = map(lambda line: line.split("\t")[:2], f.readlines())
+    for image_filename, image_label in val_annotation_data:
+        from_image_filepath = val_images_dir / image_filename
+        to_image_dir = val_data_dir / image_label
+        if not to_image_dir.exists():
+            to_image_dir.mkdir()
+        to_image_filepath = to_image_dir / image_filename
+        from_image_filepath.rename(to_image_filepath)
+    val_annotations_file.unlink()
+    val_images_dir.rmdir()
+
+
+def main():
+    device = torch.device("cpu")
+    print(f"Using {device} device")
+
+    ###############################################################################
+    # Step 1: Prepare model and dataset
+    print(os.linesep + "[Step 1] Prepare model and dataset")
+
+    model = get_resnet18_model(device)
+    model, acc1_fp32 = load_checkpoint(model)
+
+    print(f"Accuracy@1 of original FP32 model: {acc1_fp32}")
+
+    val_loader, calibration_dataset = create_data_loaders()
+
+    def transform_fn(data_item):
+        return data_item[0].to(device)
+
+    quantization_dataset = nncf.Dataset(calibration_dataset, transform_fn)
+
+    ###############################################################################
+    # Step 2: Quantize model
+    print(os.linesep + "[Step 2] Quantize model")
+
+    input_shape = (1, 3, IMAGE_SIZE, IMAGE_SIZE)
+    example_input = torch.ones(*input_shape).cpu()
+
+    fx_model = torch.export.export(model.eval(), args=(example_input,)).module()
+    quantized_fx_model = nncf.quantize(fx_model, quantization_dataset)
+
+    quantized_fx_model = torch.compile(quantized_fx_model, backend="openvino")
+
+    acc1_int8 = validate(val_loader, quantized_fx_model, device)
+    print(f"Accuracy@1 of INT8 model: {acc1_int8:.3f}")
+
+    ###############################################################################
+    # Step 3: Run benchmarks
+    print(os.linesep + "[Step 3] Run benchmarks")
```

Comment:
I would suggest you explicitly write that you are testing torch.compile with the OpenVINO backend.

Reply:
Done, please check